### PR TITLE
tests: Adds guestbook app subcommand in agnhost

### LINKED ISSUE #80534: Test: Guestbook application should create and stop a working app fails on Windows
Failing on: 

- aks-engine-windows-master-staging
- aks-engine-windows-master
- gce-windows-master
- flannel-*-windows-master

Which test:

[sig-cli] Kubectl client [k8s.io] Guestbook application should create and stop a working application [Conformance]

**Testgrid link**:

https://testgrid.k8s.io/sig-windows#aks-engine-azure-windows-master-staging

**Reason for failure**:
```

Jul 23 23:48:34.261: INFO: Failed to get response from guestbook. err: <nil>, response: <br />
<b>Fatal error</b>:  Uncaught Predis\Response\ServerException: DENIED Redis is running in protected mode because protected mode is enabled, no bind address was specified, no authentication password is requested to clients. In this mode connections are only accepted from the loopback interface. If you want to connect from external computers to Redis you may adopt one of the following solutions: 1) Just disable protected mode sending the command 'CONFIG SET protected-mode no' from the loopback interface by connecting to Redis from the same host the server is running, however MAKE SURE Redis is not publicly accessible from internet if you do so. Use CONFIG REWRITE to make this change permanent. 2) Alternatively you can just disable the protected mode by editing the Redis configuration file, and setting the protected mode option to 'no', and then restarting the server. 3) If you started the server manually just for testing, restart it with the '--protected-mode no' option. 4) Setup a bind address or an authentication password. NOTE:  in <b>C:\Users\ContainerAdministrator\AppData\Roaming\Apache24\htdocs\Predis\src\Client.php</b> on line <b>370</b><br />
```

**Anything else we need to know**:

Most probably related to the recent transition to redis:5.0.5. Windows still uses a Redis port that is version 3.2 . Since there is no Windows version newer than 3.2 for windows we need to investigate how to accommodate the test within the existing image.

### PR DESCRIPTION
**Pull request by claudiubelu**
**What type of PR is this?**

/kind failing-test

/sig windows
/sig testing

/area conformance

**What this PR does / why we need it**:

The redis version has been bumped to version 5.0.5, but the maximum version supported on Windows is 3.2. This can lead to failing tests, the output and behaviour can be different (see #80516). In order to prevent such failures, the amount of times the Redis image is used can be reduced.

This commit adds the guestbook subcommand to agnhost, which can be used to emulate the Guestbook application created by the test ``should create and stop a working application``.

Bumps agnhost image VERSION.

**Which issue(s) this PR fixes**:

Partially fixes #80534

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by alenkacz** [comment]
@bclau any update on this PR? Seems to be failing a lot of tests. I am just pinging because it's related to a test failure we're observing https://github.com/kubernetes/kubernetes/issues/80534

**Comment by bclau** [comment]
> 
> 
> @bclau any update on this PR? Seems to be failing a lot of tests. I am just pinging because it's related to a test failure we're observing #80534

The reason for failure is that the ``agnhost`` image version was bumped, but it was not built and published yet.

File: test/images/agnhost/guestbook/guestbook.go [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+			continue
+		}
+
+		responseJSON := make(map[string]interface{})
+		err = json.Unmarshal([]byte(response), &responseJSON)
+		if err != nil {
+			log.Fatalf("Error while unmarshaling master's response: %v", err)
+		}
+
+		store = responseJSON["data"].(map[string]interface{})
```
Comment (tedyu) [inline test/images/agnhost/guestbook/guestbook.go]: Should we check whether the cast doesn't have error ?

Comment (bclau) [reply test/images/agnhost/guestbook/guestbook.go]: Done

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

**Comment by listx** [comment]
I will now try to publish the images in here.

**Comment by listx** [comment]
Published:
```
./image-util.sh push agnhost
The push refers to repository [gcr.io/kubernetes-e2e-test-images/agnhost-amd64]
c6b2eb29f1d1: Pushed 
0afb5422fc41: Pushed 
8875cdb6cc91: Pushed 
d05e0336313b: Pushed 
da80e910f5d9: Pushed 
8c8a4c0c2344: Pushed 
721384ec99e5: Layer already exists 
2.8: digest: sha256:bff16769cddd9827d8040952b4255ebe839b825ed9cb58cdf2dc988f5191f0f8 size: 1779
The push refers to repository [gcr.io/kubernetes-e2e-test-images/agnhost-arm]
eb37068e2a79: Pushed 
cace34442629: Pushed 
da240bf6f894: Pushed 
bdd177597149: Pushed 
39ead5522a91: Pushed 
1c6334cdb400: Pushed 
fcf3452f08cb: Pushed 
845b5c25f81e: Layer already exists 
2.8: digest: sha256:933fe0a6227cf6261b1f65b0bdfb40789958d3138fd308759b7b0520219832e8 size: 1990
The push refers to repository [gcr.io/kubernetes-e2e-test-images/agnhost-arm64]
c784e64885fa: Pushed 
ebb22e227b1a: Pushed 
2f2fdaf748be: Pushed 
ed3a7d7e8769: Pushed 
5d176df10f08: Pushed 
0511a5781f33: Pushed 
614e7540d8ec: Pushed 
56e86ee0b452: Layer already exists 
2.8: digest: sha256:79f6b61ccd82bd8dabc28740c815a1b492faefe3883e00b68c39c54e0ab0ed93 size: 1989
The push refers to repository [gcr.io/kubernetes-e2e-test-images/agnhost-ppc64le]
feed0c67d2b5: Pushed 
ddfee4810399: Pushed 
8841feb29c44: Pushed 
d76d119eb001: Pushed 
34406e5986b5: Pushed 
51af9f443f1e: Pushed 
15b74036d530: Pushed 
85e899b8f0cc: Layer already exists 
2.8: digest: sha256:9060e99afd0ddaac91d8e6e4da94f6a2642507a5b4baaa1295db398c640cd154 size: 1989
The push refers to repository [gcr.io/kubernetes-e2e-test-images/agnhost-s390x]
3757e098c807: Pushed 
c6acdebe8f6c: Pushed 
3beefbbf6e39: Pushed 
ae5feaf7efe4: Pushed 
857b51a7e2c2: Pushed 
e74cfb838b18: Pushed 
50e1c50e10c3: Pushed 
f129ff7d977b: Layer already exists 
2.8: digest: sha256:12f0a9fce29d5567b6e13887f7633a104bcbad150a503c308bbb3d2f5f177c2a size: 1986
Created manifest list gcr.io/kubernetes-e2e-test-images/agnhost:2.8
Pushed ref gcr.io/kubernetes-e2e-test-images/agnhost@sha256:bff16769cddd9827d8040952b4255ebe839b825ed9cb58cdf2dc988f5191f0f8 with digest: sha256:bff16769cddd9827d8040952b4255ebe839b825ed9cb58cdf2dc988f5191f0f8
Pushed ref gcr.io/kubernetes-e2e-test-images/agnhost@sha256:933fe0a6227cf6261b1f65b0bdfb40789958d3138fd308759b7b0520219832e8 with digest: sha256:933fe0a6227cf6261b1f65b0bdfb40789958d3138fd308759b7b0520219832e8
Pushed ref gcr.io/kubernetes-e2e-test-images/agnhost@sha256:79f6b61ccd82bd8dabc28740c815a1b492faefe3883e00b68c39c54e0ab0ed93 with digest: sha256:79f6b61ccd82bd8dabc28740c815a1b492faefe3883e00b68c39c54e0ab0ed93
Pushed ref gcr.io/kubernetes-e2e-test-images/agnhost@sha256:9060e99afd0ddaac91d8e6e4da94f6a2642507a5b4baaa1295db398c640cd154 with digest: sha256:9060e99afd0ddaac91d8e6e4da94f6a2642507a5b4baaa1295db398c640cd154
Pushed ref gcr.io/kubernetes-e2e-test-images/agnhost@sha256:12f0a9fce29d5567b6e13887f7633a104bcbad150a503c308bbb3d2f5f177c2a with digest: sha256:12f0a9fce29d5567b6e13887f7633a104bcbad150a503c308bbb3d2f5f177c2a
sha256:39f79e4b287745553c31afe1c8f3790cb4df3366e9084389f0d6603dadd25c46
```

**Comment by droslean** [comment]
This PR didn't fix the problem. The job is still failing.
https://testgrid.k8s.io/sig-release-master-informing#gce-windows-master

**Comment by bclau** [comment]
> 
> 
> This PR didn't fix the problem. The job is still failing.
> https://testgrid.k8s.io/sig-release-master-informing#gce-windows-master
> 
> @listx Can you please check it?

This is a 2-part fix. One is this PR, one is the one that consumes what this PR did: https://github.com/kubernetes/kubernetes/pull/84046

But the agnhost image needs to be rebuilt first (because of this: https://github.com/kubernetes/kubernetes/pull/84825)

**Comment by listx** [comment]
Published again (this pulls in #84825):

```
./image-util.sh push agnhost
The push refers to repository [gcr.io/kubernetes-e2e-test-images/agnhost-amd64]
f5a613a7dcb1: Pushed 
d605813fa297: Pushed 
8875cdb6cc91: Layer already exists 
d05e0336313b: Layer already exists 
da80e910f5d9: Layer already exists 
8c8a4c0c2344: Layer already exists 
721384ec99e5: Layer already exists 
2.8: digest: sha256:02b9e10ea1f7439a03d9bc3ee16d984172ca7eac1818792ad4d721c5c8f72ff4 size: 1778
The push refers to repository [gcr.io/kubernetes-e2e-test-images/agnhost-arm]
cec31dd067d7: Pushed 
d6b17c44ff88: Pushed 
da240bf6f894: Layer already exists 
bdd177597149: Layer already exists 
39ead5522a91: Layer already exists 
1c6334cdb400: Layer already exists 
fcf3452f08cb: Layer already exists 
845b5c25f81e: Layer already exists 
2.8: digest: sha256:83c4182ef79ce80448ea21d4406c75c511d071cb18f30fa06cee17b940d5bc5c size: 1989
The push refers to repository [gcr.io/kubernetes-e2e-test-images/agnhost-arm64]
f3ba36831d95: Pushed 
477485df31f3: Pushed 
2f2fdaf748be: Layer already exists 
ed3a7d7e8769: Layer already exists 
5d176df10f08: Layer already exists 
0511a5781f33: Layer already exists 
614e7540d8ec: Layer already exists 
56e86ee0b452: Layer already exists 
2.8: digest: sha256:a68d76cee2471a8806d62bca76d19a3c9c86bfd9e4a75cbbf6b03927e5eebce4 size: 1989
The push refers to repository [gcr.io/kubernetes-e2e-test-images/agnhost-ppc64le]
23617a06badd: Pushed 
de7d36104a5d: Pushed 
8841feb29c44: Layer already exists 
d76d119eb001: Layer already exists 
34406e5986b5: Layer already exists 
51af9f443f1e: Layer already exists 
15b74036d530: Layer already exists 
85e899b8f0cc: Layer already exists 
2.8: digest: sha256:7c151ccbfa7461738860a8b2920a12c925751121d3f24be2fbfb2c96b4173611 size: 1989
The push refers to repository [gcr.io/kubernetes-e2e-test-images/agnhost-s390x]
8bf312344659: Pushed 
9710d9e7e85f: Pushed 
3beefbbf6e39: Layer already exists 
ae5feaf7efe4: Layer already exists 
857b51a7e2c2: Layer already exists 
e74cfb838b18: Layer already exists 
50e1c50e10c3: Layer already exists 
f129ff7d977b: Layer already exists 
2.8: digest: sha256:78fc7e14d7b59a590bd802517c44435969ceff5baedf2791604c2f0576a65c19 size: 1985
Created manifest list gcr.io/kubernetes-e2e-test-images/agnhost:2.8
Pushed ref gcr.io/kubernetes-e2e-test-images/agnhost@sha256:02b9e10ea1f7439a03d9bc3ee16d984172ca7eac1818792ad4d721c5c8f72ff4 with digest: sha256:02b9e10ea1f7439a03d9bc3ee16d984172ca7eac1818792ad4d721c5c8f72ff4
Pushed ref gcr.io/kubernetes-e2e-test-images/agnhost@sha256:83c4182ef79ce80448ea21d4406c75c511d071cb18f30fa06cee17b940d5bc5c with digest: sha256:83c4182ef79ce80448ea21d4406c75c511d071cb18f30fa06cee17b940d5bc5c
Pushed ref gcr.io/kubernetes-e2e-test-images/agnhost@sha256:a68d76cee2471a8806d62bca76d19a3c9c86bfd9e4a75cbbf6b03927e5eebce4 with digest: sha256:a68d76cee2471a8806d62bca76d19a3c9c86bfd9e4a75cbbf6b03927e5eebce4
Pushed ref gcr.io/kubernetes-e2e-test-images/agnhost@sha256:7c151ccbfa7461738860a8b2920a12c925751121d3f24be2fbfb2c96b4173611 with digest: sha256:7c151ccbfa7461738860a8b2920a12c925751121d3f24be2fbfb2c96b4173611
Pushed ref gcr.io/kubernetes-e2e-test-images/agnhost@sha256:78fc7e14d7b59a590bd802517c44435969ceff5baedf2791604c2f0576a65c19 with digest: sha256:78fc7e14d7b59a590bd802517c44435969ceff5baedf2791604c2f0576a65c19
sha256:daf5332100521b1256d0e3c56d697a238eaec3af48897ed9167cbadd426773b5```

**Pull request merged**